RecoEgamma/EgammaTools/src/SCEnergyCorrectorSemiParm.cc

```diff
@@ -294,9 +297,22 @@ void SCEnergyCorrectorSemiParm::modifyObject(reco::SuperCluster &sc) {
     double ecor = mean*eval[6];
     if (!iseb)  
       ecor = mean*eval[6]+sc.preshowerEnergy();
+
+	p.first  = ecor;
+	p.second = 0.;
+
+  }
+
+  return p;
+}
    
-    sc.setEnergy(ecor);
-    sc.setCorrectedEnergy(ecor);
-  }  
+//--------------------------------------------------------------------------------------------------
+void SCEnergyCorrectorSemiParm::modifyObject(reco::SuperCluster &sc) {
+  
+	std::pair<double, double> cor = getCorrections(sc);
+	if(cor.first<0 || cor.second < 0) return;
+	sc.setEnergy(cor.first);
+	sc.setCorrectedEnergy(cor.first);
+	sc.setCorrectedEnergyUncertainty(cor.second);
```

Comment:
default ("uninitialized") energy uncertainty is -1.
I think we better keep it like that  at -1 and not change it to 0 for cases where we think the value is not used.